src/Caliburn.Micro.Platform/Platforms/netcore-avalonia/NavigationFrame.cs

```diff
@@ -117,17 +116,25 @@ private void NavigateToViewModel(object viewModel)
 
             ViewModelBinder.Bind(viewModel, viewInstance, null);
             Log.Info($"View Model {viewModel}");
-            Log.Info($"View {viewInstance}");
             Tag = "Navigating";
             viewInstance.DataContext = viewModel;
             Content = viewInstance;
-
+            if (addToStack)
+            {
+                navigationStack.Add(viewModel);
+                navigationStackIndex = navigationStack.Count;
+            }
         }
 
         /// <inheritdoc/>
         public Task GoBackAsync(bool animated = true)
         {
-            throw new NotImplementedException();
+            Log.Info("Going back");
+            if (navigationStackIndex > 0)
+                navigationStackIndex--;
+
+            NavigateToViewModel(navigationStack[navigationStackIndex - 1], false);
```

Comment:
If `navigationStackIndex` is 1 and and we go through here will first decrement `navigationStackIndex` and then subtract 1 this will give an ArgumentOutOfRange exception. 
But anyway should we decrement twice? Will that not skip a place in the stack?

Reply:
fixed